src/main/java/mil/dds/anet/search/mssql/MssqlTaskSearcher.java

```diff
@@ -1,160 +1,77 @@
 package mil.dds.anet.search.mssql;
 
-import com.google.common.base.Joiner;
-import java.util.HashMap;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
 import mil.dds.anet.beans.Task;
-import mil.dds.anet.beans.lists.AnetBeanList;
 import mil.dds.anet.beans.search.ISearchQuery.SortOrder;
 import mil.dds.anet.beans.search.TaskSearchQuery;
-import mil.dds.anet.beans.search.TaskSearchQuery.TaskSearchSortBy;
-import mil.dds.anet.database.mappers.TaskMapper;
-import mil.dds.anet.search.AbstractSearcherBase;
-import mil.dds.anet.search.ITaskSearcher;
-import mil.dds.anet.utils.DaoUtils;
+import mil.dds.anet.search.AbstractSearchQueryBuilder;
+import mil.dds.anet.search.AbstractTaskSearcher;
 import mil.dds.anet.utils.Utils;
-import org.jdbi.v3.core.statement.Query;
 
-public class MssqlTaskSearcher extends AbstractSearcherBase implements ITaskSearcher {
+public class MssqlTaskSearcher extends AbstractTaskSearcher {
+
+  public MssqlTaskSearcher() {
+    super(new MssqlSearchQueryBuilder<Task, TaskSearchQuery>("MssqlTaskSearch"));
+  }
 
   @Override
-  public AnetBeanList<Task> runSearch(TaskSearchQuery query) {
-    final List<String> whereClauses = new LinkedList<String>();
-    final Map<String, Object> args = new HashMap<String, Object>();
-    final StringBuilder sql = new StringBuilder("/* MssqlTaskSearch */ SELECT tasks.*");
+  protected void buildQuery(TaskSearchQuery query) {
+    super.buildQuery(query);
+    qb.addSelectClause("COUNT(*) OVER() AS totalCount");
```

Comment:
Can't we do the totalcount only in one place, for all mssql queries? Or add an addTotalCount method we could call here?

Reply:
In some cases the count needs ot be done in the outer query, so we can't always add it to the inner query. But a generic `addTotalCount(qb)` method seems like a good idea.